### ISSUE TITLE
Error Using Collapse Without Unit

### ISSUE THREAD

**Issue by JSRankins** [cqframework/cql-execution#255]
The [CumulativeMedicationDuration](https://github.com/cqframework/CQL-Formatting-and-Usage-Wiki/blob/master/Source/Libraries/CumulativeMedicationDuration-0.3.000.cql) library contains a function using a collapse.
```
define function "CumulativeDuration"(Intervals List<Interval<Date>> ):
  Sum((collapse Intervals) X return all difference in days between start of X and end of X)
```
This function throws an error when more than one interval is in the list (e.g., {Interval[@2010-01-01, @2010-06-01], Interval[@2010-06-02, @2010-12-31]} ) provided to the function. Stacktrace:

```
cqm_calculator.self.js?body=1:75 Error: Cannot create a quantity with an undefined value
    at new Quantity (browser.self.js?body=1:24952)
    at Interval.getPointSize (browser.self.js?body=1:24691)
    at collapseIntervals (browser.self.js?body=1:29895)
    at Collapse.exec (browser.self.js?body=1:29856)
    at Collapse.execute (browser.self.js?body=1:28530)
    at MultiSource.forEach (browser.self.js?body=1:32670)
    at Query.exec (browser.self.js?body=1:32538)
    at Query.execute (browser.self.js?body=1:28530)
    at Sum.exec (browser.self.js?body=1:25554)
    at Sum.execute (browser.self.js?body=1:28530)
    at FunctionRef.exec (browser.self.js?body=1:32980)
    at FunctionRef.execute (browser.self.js?body=1:28530)
    at browser.self.js?body=1:28548
    at Array.map (<anonymous>)
    at Greater.execArgs (browser.self.js?body=1:28547)
```

CQL supports the use of [collapse](https://cql.hl7.org/09-b-cqlreference.html#collapse) with or without a specified unit for the collapse. The function in the library does not specify a unit. Given the stacktrace, we speculate that this is the issue but question if that behavior is correct based on CQL.

**Comment by cmoesel** [comment]
@JSRankins - I just took a look at our code and we do have several unit tests that specifically test collapse of DateTime intervals _without_ specifying a per unit.  The tests all pass -- so there is some edge case happening here.  I'll look into it further -- but it would help to know if there is anything unusual about the intervals being passed in.  Are any of them null?  Do any have null boundaries?  If so, open null or closed null?  Etc...

**Comment by JSRankins** [comment]
That confirms what we were seeing @cmoesel. When per is added to the CQL logic, https://github.com/cqframework/cql-execution/issues/256 arises.

**Comment by JSRankins** [comment]
Tagging @brynrhodes on this since this issue impacts the QDM version of CumulativeMedicationDuration library. Bryn, sounds like @cmoesel is working on updating the code, but depending on timing for release and when something might be able to get into Bonnie, thoughts about using a workaround for this logic in the CMD library?

**Comment by cmoesel** [comment]
This fix is now available in the cql-execution 2.3.3 release.
